### PR TITLE
fix bug #1164

### PR DESCRIPTION
**Pull request by hanfei19910905**
fix bug #1164

### CODE REVIEW

**Comment by hanfei19910905** [comment]
@shenli @coocood PTAL

File: optimizer/plan/planbuilder_join.go [hunk 1]
```diff
@@ -209,7 +209,7 @@ func (p *joinPath) attachCondition(condition ast.ExprNode, availablePaths []*joi
 		p.filterRate *= filterRate
 		return true
 	}
-	if p.inner.attachCondition(condition, append(availablePaths, p.outer)) {
+	if p.inner.attachCondition(condition, availablePaths) {
```
Comment (coocood) [inline optimizer/plan/planbuilder_join.go]: If this is an ON condition, we should push it down to inner with `p.outer` as availablePath.

Comment (hanfei19910905) [inline optimizer/plan/planbuilder_join.go]: It will never be a on condition. On condition will be processed in newOuterJoinPath.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
I tried this case and it failed.

`select * from t left outer join t1 on t.c1 = t1.c1 where t1.c1 = 3 or false`

expected is empty result, but got
```
+------+------+------+------+
| c1   | c2   | c1   | c2   |
+------+------+------+------+
|    1 |    1 | NULL | NULL |
|    2 |    2 | NULL | NULL |
+------+------+------+------+
```

**Comment by coocood** [comment]
If the condition is from where, it should never attach to an inner path.

**Comment by coocood** [comment]
and this case 
`select * from t left outer join t1 on t.c1 = t1.c1 where t1.c1 != 2 or t1.c1 is true`

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**